src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -92,6 +96,25 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 c =>
                 {
                     var docOptions = document.GetOptionsAsync(c.CancellationToken).WaitAndGetResult(c.CancellationToken);
+                    var workspaceOptions = document.Project.Solution.Workspace.Options;
+
+                    // AB Test
+                    // If the feature is OFF and the user has not previously been assigned to a test group, assign them now
+                    if (!docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting)
+                        && workspaceOptions.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment) == false)
+                    {
+                        var experimentationService = document.Project.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
```

Comment:
>GetRequiredService [](start = 98, length = 18)

`GetRequiredService` will throw if the experimentation service is null. We should probably just use `GetService()`. If we really want to require the service, the null check below can be removed.